Kudu.Services/ServiceHookHandlers/GithubHandler.cs

```diff
@@ -17,6 +36,25 @@ public override DeployAction TryParseDeploymentInfo(HttpRequestBase request, JOb
             return DeployAction.UnknownPayload;
         }
 
+        protected override string determineSecurityProtocol(JObject repository)
+        {
+            string repositoryUrl = repository.Value<string>("url");
+            bool isPrivate = repository.Value<bool>("private");
+            if (isPrivate)
+            {
+                repositoryUrl = repository.Value<string>("ssh_url");
+            }
+            else if (!String.Equals(new Uri(repositoryUrl).Host, "github.com"))
+            {
+                // github enterprise user, we don't really need to check for null...since only test will invoke githubhandler
```

Comment:
This comment is not so meaningful to me. I would instead describe what the code is doing so someone reading it knows why we're doing this, e.g.

    // If we find an SSH key and it's GitHub Enterprise and not github.com, use ssh. This is needed if
    // the repo is public but the GHE instance is not